SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/ReferenceCountedLockingPolicy.cpp

```diff
@@ -62,11 +63,15 @@ ActorLockToken UReferenceCountedLockingPolicy::AcquireLock(AActor* Actor, FStrin
 	{
 		// We want to avoid memory leak if a locked actor is deleted.
 		// To do this, we register with the Actor OnDestroyed delegate with a function that cleans up the internal map.
-		Actor->OnDestroyed.AddDynamic(this, &UReferenceCountedLockingPolicy::OnLockedActorDeleted);
-		ActorToLockingState.Add(Actor, MigrationLockElement{ 1, [this, Actor]
+		if (!Actor->OnDestroyed.IsAlreadyBound(this, &UReferenceCountedLockingPolicy::OnExplicitlyLockedActorDeleted))
```

Comment:
If this is only done the first time the lock is created, when would this already be bound?

Reply:
You can have multiple game systems locking the same Actor